gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -333,35 +344,35 @@ public void onSelectPhotoResult(Uri uri) {
       return;
     }
     try {
-      Bitmap bitmap = bitmapUtil.fromUri(uri);
-      onPhotoResult(bitmap);
+      onPhotoResult(
+          PhotoResult.createSelectResult(
+              requireNonNull(fieldWaitingForPhoto), bitmapUtil.fromUri(uri)));
       Timber.v("Select photo result returned");
     } catch (IOException e) {
       Timber.e(e, "Error getting photo returned by camera");
     }
   }
 
-  public void onCapturePhotoResult(Bitmap thumbnail) {
-    if (thumbnail == null) {
-      Timber.e("onCapturePhotoResult called with null thumbnail");
+  public void onCapturePhotoResult(Boolean result) {
+    if (result == null || !result) {
+      Timber.e("onCapturePhotoResult called with false result");
```

Comment:
Is a false result necessarily an error, or could it also mean the user dismissed the capture flow? Please add a comment explaining what `result` refers too.

Reply:
Done